lib/logpipe.h

```diff
@@ -221,6 +222,8 @@ struct _LogPipe
   gboolean (*init)(LogPipe *self);
   gboolean (*deinit)(LogPipe *self);
 
+  const gchar *(*generate_persist_id)(LogPipe *self);
```

Comment:
I'd prefer generate_persist_name, since returned value is usually prefixed by the driver.